service/src/main/kotlin/fi/espoo/evaka/titania/Titania.kt

```diff
@@ -0,0 +1,248 @@
+// SPDX-FileCopyrightText: 2021-2022 City of Tampere
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+package fi.espoo.evaka.titania
+
+import com.fasterxml.jackson.annotation.JsonFormat
+import com.fasterxml.jackson.annotation.JsonProperty
+import fi.espoo.evaka.shared.domain.FiniteDateRange
+import java.time.LocalDate
+import org.springframework.http.HttpStatus
+
+internal const val TITANIA_DATE_FORMAT = "yyyyMMdd"
+internal const val TITANIA_TIME_FORMAT = "HHmm"
+
+// from Titania_Vakiotyöaikatapahtumakoodit_Julkinen.pdf v1.3
+
+internal val IGNORED_EVENT_CODES =
+    listOf(
+        // Työrupeamat
+        "(",
+        "£",
+        "&",
+        "€",
+        // Vapaapäivät
+        "V",
+        "L",
+        ":",
+        "/",
+        ")",
+        "%",
+        "T",
+        "!",
+        "k",
+        "Q",
+        "q",
+        "v",
+        "+",
+        // Koulutukset
+        "N",
+        "n",
+        // Varallaolo ja hälytystyö
+        "1",
+        "2",
+        "3",
+        "4",
+        "5",
+        "9",
+        "8",
+        // Vuosilomat
+        "H",
+        "h",
+        // Sairauslomat
+        "S",
+        "s",
+        "Z",
+        // Vanhempainvapaat
+        "7",
+        "G",
+        "g",
+        // Virkavapaat
+        "M",
+        "m",
+        // Muut poissaolot
+        "t",
+        "#",
+        "l",
+        "{",
+        "\\",
+        "[",
+        // Muut tapahtumat
+        ".",
+        "*",
+        "@",
+        "<"
+    )
+
+// from updateWorkingTimeEvents.wsdl, version 1.2 25.8.2020 & getStampedWorkingTimeEvents.wsdl,
+// version 1.1 14.8.2020
+
+data class TitaniaCode(val code: String, val name: String? = null)
+
+data class TitaniaCodeName(val code: String, val name: String)
+
+data class TitaniaPeriod(
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = TITANIA_DATE_FORMAT)
+    val beginDate: LocalDate,
+    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = TITANIA_DATE_FORMAT)
+    val endDate: LocalDate
+) {
+    fun toDateRange() = FiniteDateRange(beginDate, endDate)
+
+    companion object {
+        fun from(range: FiniteDateRange) = TitaniaPeriod(range.start, range.end)
+
+        fun from(date: LocalDate) = TitaniaPeriod(date, date)
+    }
+}
+
+enum class TitaniaPayrollItemType {
+    /** Edelliseltä jaksolta siirtyvää */
+    PREVIOUS,
+
+    /** Kuluva jakso */
+    THIS,
+
+    /** Seuraavalle siirtyvää */
+    NEXT,
+
+    /** Aikanakorvattavat (aikahyvitys) */
+    TIME,
+
+    /** Rahana korvattavat */
+    MONEY
+}
+
+enum class TitaniaPayrollItemUnit {
+    /** Minuutit */
+    MINUTE,
+
+    /** Kappaleet */
+    QUANTITY
+}
+
+data class UpdateWorkingTimeEventsRequest(
```

Comment:
Add `@ExcludeCodeGen` annotation to all route payload / return types to not have them included in `frontend/src/lib-common/generated/api-types/`
```suggestion
@ExcludeCodeGen
data class UpdateWorkingTimeEventsRequest(
```